### PR TITLE
Optional listname for push and pull functions

### PR DESCRIPTION
**Pull request by pascalfleury**
This makes scripting these calls easier, rather than going one level down and using more "internal" functions.

This is very useful when you want to bind this in a call from a timer to update the tests regularly (e.g. when refreshing the calendar view).

### CODE REVIEW

File: org-gtasks.el [hunk 1]
```diff
@@ -454,13 +456,15 @@
...
 
-(defun org-gtasks-push (account)
+(defun org-gtasks-push (account &optional listname)
   (let* ((tasklists (org-gtasks-tasklists account))
 	 (collection (mapcar 'tasklist-title tasklists))
 	 (collection (if (> (length collection) 1)
                          (append (list "ALL") collection)
                        collection))
-	 (target (completing-read "Push: " collection)))
+	 (target (if (eq listname nil)
```
Comment (JulienMasson) [inline org-gtasks.el:465]: can you please also replace `(eq listname nil)` by `(null listname)` here

Comment (JulienMasson) [inline org-gtasks.el:465]: can you please also replace `(eq listname nil)` by `(null listname)` here

File: org-gtasks.el [hunk 2]
```diff
@@ -454,13 +456,15 @@
...
+                     (completing-read "Push: " collection)
```
Comment (JulienMasson) [inline org-gtasks.el:466]: can you please also replace `(eq listname nil)` by `(null listname)` here

File: org-gtasks.el (hunk 1 above)
Comment (JulienMasson) [inline org-gtasks.el:465]: can you please also replace `(eq listname nil)` by `(null listname)` here

Comment (JulienMasson) [inline org-gtasks.el]: can you please also replace `(eq listname nil)` by `(null listname)` here

**Comment by pascalfleury** [comment]
somehow anything I do with git will end up with an even more messed up history...

**Comment by JulienMasson** [comment]
your master is not aligned with mine, that's why when you create Pull-Request you have unwanted commits.

You should do something like this to align your master:
``` shell
$ git remote add -f upstream https://github.com/JulienMasson/org-gtasks.git
$ git push --force origin 'upstream/master:refs/heads/origin/master'
```

**Comment by pascalfleury** [comment]
That was one reason. But you have not seen the mess I made in my git
history :-)
I went ahead and read most of the git book, as I was basically dealing with
remote repositories completely wrong.
Thanks for your patience !
--paf

**Comment by JulienMasson** [comment]
> But you have not seen the mess I made in my git history :-)

ahahaha indeed

> I went ahead and read most of the git book, as I was basically dealing with
remote repositories completely wrong.
Thanks for your patience !

no problem :)

**Pull request closed without merging**